GeometryService/src/ProtonBeamDumpMaker.cc

```diff
@@ -20,224 +21,308 @@
-      dump->_backShieldingOutline.emplace_back(pswVertices[2][0]+psWoffset[2]+deltaSW[0],pswVertices[2][1]+psWoffset[0]+deltaSW[1]);
-    }
-      // Due to changes in bldg construction, use vertex [7] instead of [2]
-    dump->_backShieldingOutline.emplace_back(psSVertices[7][0]+psSoffset[2]+deltaSE[0],psSVertices[7][1]+psSoffset[0]+deltaSE[1]);
-    dump->_backShieldingOutline.emplace_back(zs,xs);
-    dump->_backShieldingOutline.emplace_back(zn,xn);
-
-    if ( version > 2 ) {
-      dump->_frontShieldingOutline.emplace_back(psNVertices[5][0]+psNoffset[2]+deltaNE[0],psNVertices[5][1]+psNoffset[0]+deltaNE[1]);
-    } else {
-      dump->_frontShieldingOutline.emplace_back(psNVertices[7][0]+psNoffset[2]+deltaNE[0],psNVertices[7][1]+psNoffset[0]+deltaNE[1]);
-    }
+    // orientation and the nominal dump rotation.
+
+    // The vertical extent of beam dump concrete is from psArea concrete to the ceiling.
+    const double concreteYmin = psArea.getOffsetFromMu2eOrigin().y() + psArea.getYhalfThickness();
+    const double concreteYmax = psCeil.getOffsetFromMu2eOrigin().y() - psCeil.getYhalfThickness();
+
+    dump->_dumpConcreteHalfHeight = (concreteYmax - concreteYmin)/2;
+    dump->_dumpConcreteCenterInMu2e = Hep3Vector(0, (concreteYmax+concreteYmin)/2, 0);
+
+    // Beam dump outline is contained by psWallUpper vertices 13-14-15-16,
+    // with the back of the dump following the 14-15 line.
+    // The front corners are at the intersections of straight lines
+    // through psWallUpper vertices 13-14 or 15-16 and the beam dump face.
+
+    const double dumpCoreCenterToConcreteFaceDistance =
+      dump->_coreHalfSize[2]
+      + 2 * dump->_neutronCaveHalfSize[2]
+      + 2*dump->_mouthHalfSize[2];
+
+    // Beam dump face line in the plan view
+    Line faceLine(
+                  // Concrete face point on dump axis
+                  toOutlinePlane(dump->beamDumpToMu2e_position(Hep3Vector(0, 0, dumpCoreCenterToConcreteFaceDistance))),
+                  // Move 10m sideway.  An arbitrary point to define the line.
+                  toOutlinePlane(dump->beamDumpToMu2e_position(Hep3Vector(10000., 0, dumpCoreCenterToConcreteFaceDistance)))
+                  );
+
+    const auto& pswVertices =  psWall.getVertices();
+    const Hep2Vector& psWoff2 = toOutlinePlane(psWall.getOffsetFromMu2eOrigin());
+
+    dump->_dumpConcreteOutline.emplace_back(pswVertices[14]+psWoff2 + ovlgap*(-nx+nz));
+    dump->_dumpConcreteOutline.emplace_back(pswVertices[15]+psWoff2 + ovlgap*( nx+nz));
+    dump->_dumpConcreteOutline.emplace_back(intersection(faceLine, Line(pswVertices[15] + psWoff2, pswVertices[16] + psWoff2)) + ovlgap*( nx));
+    dump->_dumpConcreteOutline.emplace_back(intersection(faceLine, Line(pswVertices[13] + psWoff2, pswVertices[14] + psWoff2)) + ovlgap*(-nx));
+
+    // The ExtMon room subtraction.
+    // The the back is the same as the concrete back.  Add a margin for the G4 boolean.
+    const double booleanVolumeMargin = 50; // mm
```

Comment:
Is that a value that should be in the config file?

Reply:
No, this is a non-physical parameter that is only needed to deal with G4 subtleties. Adding it to a config file would only create confusion.